### PR TITLE
Some improvements for the state package testing

### PR DESCRIPTION
**Pull request by leolara**
**What type of PR is this?**

Refactor

**What does this PR do? Why is it needed?**

Avoids some code repetition in the state package testing

**Which issues(s) does this PR fix?**

Related #10047

**Other notes for review**

### CODE REVIEW

File: beacon-chain/state/testing/getters_block.go
Comment (leolara) [reply beacon-chain/state/testing/getters_block.go:15]: I am just copying what is the name of the original test: `TestBeaconState_LatestBlockHeader`

Comment (terencechain) [reply beacon-chain/state/testing/getters_block.go:15]: Then I think camel case is preferred since this is no longer an executable test rather than a test only helper function

**Comment by leolara** [comment]
@terencechain please, check new commit

**Review by rkapka — CHANGES_REQUESTED** [review]
Please also add `state-native/v2/references_test.go` and `state-native/v3/references_test.go`

**Comment by leolara** [comment]
@rkapka done

**Review by rkapka — APPROVED** [review]

**Comment by rkapka** [comment]
Thanks! Let's wait for @terencechain's approval before merging

**Comment by rkapka** [comment]
`bazel run //:gazelle -- fix` needed

**Review by rkapka — DISMISSED** [review]

**Comment by rkapka** [comment]
Hey @leolara , I just checked https://github.com/prysmaticlabs/prysm/pull/10316/commits/daaa2f9f2101c30667487cea2d3af1d926a6a3b6 and I'm pretty sure what @terencechain meant was to change `VerifyBeaconState_LatestBlockHeader` to `verifyBeaconStateLatestBlockHeader`, nothing else

**Comment by rkapka** [comment]
Thanks for explaining. This looks good to me.

**Review by rkapka — APPROVED** [review]

**Pull request merged**
